cmd/push.go

```diff
@@ -53,10 +60,72 @@ func init() {
 
 type PushCmdParams struct {
 	AccountContextParams
-	ASU         string
-	allAccounts bool
-	force       bool
-	targeted    []string
+	ASU              string
+	sysAccUser       string // when present use
+	sysAcc           string
+	sysAccUserJwtOpt nats.Option
+	allAccounts      bool
+	force            bool
+	prune            bool
+	targeted         []string
+
+	accountList []string
+}
+
+func processResponse(report *store.Report, resp *nats.Msg) (bool, string, interface{}) {
+	serverResp := map[string]interface{}{}
+	if err := json.Unmarshal(resp.Data, &serverResp); err != nil {
+		report.AddError("failed to parse response: %v data: %s", err, string(resp.Data))
+	} else if srv, ok := serverResp["server"]; !ok {
+		report.AddError("server responded without server info: %s", string(resp.Data))
+	} else if srv, ok := srv.(map[string]interface{}); !ok {
+		report.AddError("server responded with bad server info: %s", string(resp.Data))
+	} else if error, hasErr := serverResp["error"]; hasErr {
+		if error, ok := error.(map[string]interface{}); ok {
+			report.AddError("server %s responded with error: %s", srv["name"], error["description"])
+		} else {
+			report.AddError("server %s responded with bad error: %v", srv["name"], error)
+		}
+	} else if data, ok := serverResp["data"]; !ok {
+		report.AddError("server %s responded without data: %s", srv["name"], string(resp.Data))
+	} else {
+		return true, srv["name"].(string), data
+	}
+	return false, "", nil
+}
+
+func systemAccountUser(ctx ActionCtx, sysAccUserName string) (string, string, nats.Option, error) {
+	if op, err := ctx.StoreCtx().Store.ReadOperatorClaim(); err != nil {
+		return "", "", nil, err
+	} else if accNames, err := friendlyNames(ctx.StoreCtx().Operator.Name); err != nil {
+		return "", "", nil, err
+	} else if sysAccName, ok := accNames[op.SystemAccount]; !ok {
+		return "", "", nil, fmt.Errorf(`system account "%s" not found`, op.SystemAccount)
+	} else {
+		if sysAccUserName == "" {
+			sysAccUserName = "sys"
+		}
+		if claim, err := ctx.StoreCtx().Store.ReadUserClaim(sysAccName, sysAccUserName); err != nil {
+			return "", "", nil, err
+		} else if kp, err := ctx.StoreCtx().KeyStore.GetKeyPair(claim.Subject); err != nil {
+			return "", "", nil, err
+		} else if theJWT, err := ctx.StoreCtx().Store.ReadRawUserClaim(sysAccName, sysAccUserName); err != nil {
+			return "", "", nil, err
+		} else {
+			jwtCb := func() (string, error) {
+				return string(theJWT), nil
+			}
+			signCb := func(nonce []byte) ([]byte, error) {
+				return kp.Sign(nonce)
+			}
+			return sysAccName, sysAccUserName, nats.UserJWT(jwtCb, signCb), nil
+		}
+	}
+}
+
+func isHttpUrl(url string) bool {
+	return strings.HasPrefix(url, "http://") || strings.HasPrefix(url, "https://") ||
```

Comment:
this will fail unless the protocol was normalized. If this was baked, we should be trimming all white space and only testing for HasPrefix.

Reply:
fixed